### PR TITLE
grdfilter wrapper

### PR DESCRIPTION
**Pull request by carocamargo**
Added grdfilter function to the gridops.py file, and modified the .__init__.py to call the new function as well. 

Grdfilter is a wrapper of the [grdfilter](http://gmt.soest.hawaii.edu/doc/latest/grdfilter.html#limitations) function of GMT.
Given an input grid (either a datarray or a netcdf file), it applies a filter (of chosen type and length), and returns the filtered field. 
Grdfilter function was made by modifying the grdcut function.

### CODE REVIEW

**Comment by carocamargo** [comment]
I think the changes in the __init__.py were blocked. But it's just to add call gridfilter
`from .gridops import grdcut, gridfilter`

File: pygmt/__init__.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
 from .sampling import grdtrack
 from .mathops import makecpt
 from .modules import GMTDataArrayAccessor, config, info, grdinfo, which
-from .gridops import grdcut
+from .gridops import grdcut, gridfilter
```
Comment (weiji14) [inline pygmt/__init__.py:22]:
```suggestion
from .gridops import grdcut, grdfilter
```

Just a small typo, that's why it's not importing :smiley:

File: pygmt/gridops.py [hunk 2]
```diff
@@ -20,13 +20,15 @@
 
 @fmt_docstring
 @use_alias(
```
Comment (seisman) [inline pygmt/gridops.py:22]: You're changing the alias settings for `grdcut`. That's incorrect. You should revert the changes here and add decorators `@fmt_docstring`, `@use_alias` and `@kwargs_to_strings` just before the `grdfilter` function, i.e., each function has its own decorators.

File: pygmt/gridops.py [hunk 3]
```diff
@@ -113,3 +115,79 @@ def grdcut(grid, **kwargs):
...
+        
+        flag = 5: grid (x,y) in Mercator -Jm1 img units, width in km, Spherical distance calculation.
+        
+        Returns
+        -------
+        ret: xarray.DataArray or None
+        Return type depends on whether the *outgrid* parameter is set:
+        - xarray.DataArray if *outgrid* is not set
+        - None if *outgrid* is set (grid output will be stored in *outgrid*)
+        """
```
Comment (seisman) [inline pygmt/gridops.py:166]: The indentation of the docstrings looks wrong.

**Comment by weiji14** [comment]
Thanks for opening this PR @carocamargo. Once you fix the typo, you should be able to start using `pygmt.grdfilter` directly after `pip install https://github.com/carocamargo/pygmt/archive/master.zip` (and others are welcome to try too).

Let me know how much you want to work on this PR, depending on your time/Python familiarity:

Level 1. Just happy that it works, and want to get on with other stuff.
Level 2. Happy to make changes to the docstring, and maybe a couple of minor changes here and there.
Level 3. Write up a few unit tests (see example of `grdcut` PR at #492), and possibly add a gallery example.

**Review by seisman — COMMENTED** [review]
Another issue is that you're making changes to your "master" branch. It's OK but it will cause troubles after we merging this PR. It's always better to make changes in a new branch.

**Comment by carocamargo** [comment]
> Another issue is that you're making changes to your "master" branch. It's OK but it will cause troubles after we merging this PR. It's always better to make changes in a new branch.

I'm new to all github things.. This is the only way that I know how to commit the changes...

File: pygmt/gridops.py (hunk 2 above)
Comment (carocamargo) [reply pygmt/gridops.py:22]: Ok.. I thought they could be the same. Thanks!

**Comment by carocamargo** [comment]
I corrected the things as suggested by @seisman and @weiji14, and made a new PR, now from a new branch (I believe... I'm still learning to use github correctly).

**Comment by carocamargo** [comment]
> Thanks for opening this PR @carocamargo. Once you fix the typo, you should be able to start using `pygmt.grdfilter` directly after `pip install https://github.com/carocamargo/pygmt/archive/master.zip` (and others are welcome to try too).
> 
> Let me know how much you want to work on this PR, depending on your time/Python familiarity:
> 
> Level 1. Just happy that it works, and want to get on with other stuff.
> Level 2. Happy to make changes to the docstring, and maybe a couple of minor changes here and there.
> Level 3. Write up a few unit tests (see example of `grdcut` PR at #492), and possibly add a gallery example.

I wouldn't mind working on this function, even up to Level 3... but as you can see my github knowledge is quite limited

**Comment by weiji14** [comment]
> I wouldn't mind working on this function, even up to Level 3... but as you can see my github knowledge is quite limited

Excellent, we'll try to guide you along then (let us know if you have any questions), there's always a first time for everyone :smile:

I'll close this PR since you've got a parallel one open at #616.
